### DISCUSSION TITLE
OBS Dropped frames/Connections issues

### DISCUSSION THREAD

**Discussion by Amzamiviram** [owncast/owncast#3181] [General Owncast discussion]
I have owncast setup in an Ubuntu 22.04 container on Proxmox 8.  For some reason when I stream to owncast from my desktop (on the same network) I get significant frame drops/connection interruptions.(over 60% of frames dropped)  
Here are my settings in OBS.

- 6000 Kbps
- Audio Bitrate 160
- Video Encoder NVENC H.264 (also tried QSV)
- Encoder Preset Slow
- Audio Encoder AAC (Default)

Here are my stream output settings in owncast

- CPU/GPU Utilization: highest
- Resolution: 2560x1080
- 60FPS
- 5000kbps

and
- CPU/GPU Utilization: highest
- Resolution: 2560x1080
- 30FPS
- 3000kbps

Owncast never reaches more than 50% CPU or Memory, The container has access to 10 threads of an i9-12900HK.

I don't see any errors in the console when owncast is running and I shouldn't be having issues with such a low bitrate stream over my LAN.  I wouldn't think it's a transcoding issue because the dropped frames/disconnects in OBS should be happening before transcoding is done.  Any suggestions/next troubleshooting steps?

**Comment by rmens** [comment]
Drop the CPU/GPU Utilization in Owncast to lowest and see if it still occurs. I actually had this on a server that couldn't transcode the stream fast enough. It never reached full load on all threads but still couldn't handle it.

**Reply by Amzamiviram** [reply]
Oh wow, that was fast and, while I need to do more testing tomorrow it seems like that may have fixed the issue.

**Reply by rmens** [reply]
You're welcome! I would recommend running this 'stress test' file in OBS https://github.com/oszuidwest/owncast-ssl-install/blob/main/Encoder%20stress%20test%20comp%20hevc.mov and up the quality in Owncast by one level until you notice frame drops. This file contains white noise and glitter, which is a real stress test for video encoders.

I'm not sure why the bar stops at ~50% CPU, but I could reproduce this on several servers. Maybe @gabek can shed some light on this.
